src/lib/virtio/net_device.lua

```diff
@@ -196,6 +196,21 @@ function VirtioNetDevice:transmit_packets_to_vm ()
    end
 end
 
+local function validflags(buf, len)
+   local valid = checksum.verify_packet(buf, len)
+
+   if valid == true then
+      return C.VIO_NET_HDR_F_DATA_VALID
+   elseif valid == false then
+      return 0
```

Comment:
Curious: Why do we return 0 here? (What does that mean to Linux and Virtio-net, and is this better than returning `C.VIO_NET_HDR_F_NEEDS_CSUM` so that the guest will test the checksum and see the error?)